### DISCUSSION TITLE
Ideas for documentation

### DISCUSSION THREAD

**Discussion by elParaguayo** [qtile/qtile#2138] [Ideas]
I started this in the qtile-dev group but the discussions spot could be good too.

I'm looking for ideas about what documentation people would like to see added.

I've started writing a guide to creating widgets and @tych0 suggested writing up how to jail groups to particular screens.

Would love to hear ideas for what else we can improve.

To do list:
- [x] How to write a widget ([live](https://docs.qtile.org/en/latest/manual/howto/widget.html))
- [ ] How to jail a group to a screen
- [ ] How to write a layout
- [ ] Update layout docs for #1512
- [x] Add lazy.function documentation ([live](https://docs.qtile.org/en/latest/manual/config/lazy.html?highlight=lazy#user-defined-functions))
- [ ] How to write a test

I should add, volunteers are welcome to take on some of these tasks! I was just looking to collate ideas for what needs to be added. I wasn't necessarily planning to draft it all myself.

**Comment by Barbaross93** [comment]
Haha, per my recent discussion post, a writeup on how to go about writing a custom layout would be fantastic! I know that the docs talk about extensions and hooks in qtile. Personally, I haven't used the extensions and use only a couple hooks, but maybe a writeup on how to create custom extensions/hooks would be really nice?

**Comment by elParaguayo** [comment]
Thanks. I'll add the layout to the list but will need to do some research before drafting that one!

I'll take a look at the other two points too. Not familiar with extensions at all and I'm not sure you can add custom hooks (although they can be added via pull requests if there's a good use for them).

**Comment by pawamoy** [comment]
Well there still are the layout screenshots and animations that we could add to the docs: https://github.com/qtile/qtile/issues/1512 :stuck_out_tongue:

**Comment by elParaguayo** [comment]
Interesting! Would be a shame for that work to go to waste.

**Comment by pawamoy** [comment]
And you can of course get ideas from the GitHub wiki: lots of TODOs still there!

**Comment by elParaguayo** [comment]
Wow - never knew that Wiki had content in it!! I've always gone to the source code or the docs.

Edit: should the wiki be migrated to the docs so everything's in one place?

**Comment by pawamoy** [comment]
I'd be in favor of limiting the number of places where we have docs yes :sweat_smile: 
EDIT: I'm not sure about the wiki. Does it really have a place in the official docs? Not sure. Official docs should have some guides yes, but maybe not all these very specific tips and tricks. Also what's nice about the wiki is that anyone can contribute without the hassle of PRs.

**Comment by elParaguayo** [comment]
I definitely agree about the advantage of the wiki being that you don't need PRs. However, I do feel that tips and tricks absolutely have a place in the official docs (the "tips & tricks" link on the main github page even links to the "hacking" section of the docs). This is particularly true when they relate to something that people ask all the time i.e. the "Frequently Asked Questions" could cover a lot of things in the wiki.

Also, whenever I've searched for qtile info I never get results from the wiki so I didn't even know it had content. There's an obvious fix for this - just include some text in the main README file.

Just my thoughts on this.
